spec/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source_spec.rb

```diff
@@ -383,6 +383,9 @@ def playbooks_for(repo)
         expect(Notification).to receive(:create!).with(notification_args('deletion', {}))
         record.delete_in_provider
 
+        # Run most recent queue item (`GitRepository#broadcast_repo_dir_delete`)
+        MiqQueue.get.deliver
```

Comment:
I realize this is probably going to trigger @carbonin 's "this isn't a unit test" argument, but I figured removing the:

```ruby
expect(git_repo_dir).to_not exist
```

Was a worse compromise.  Mostly since while technically this is already mostly testing a lot of "Rails inner workings" when testing this method, we still want to confirm the business logic of it all is hooked up correctly, and that does require going outside of the scope of this method a bit.

However, I don't think relying on the tests from above for this is the answer either, since if we ever decide to use something besides `GitRepository` or `MiqQueue`, this spec would fail when we want it to and inform us that something in this model needs to change.  I feel like that is some "brittleness" that we can afford to confirm that this is still doing it's job when things are inevitably refactored, and save us from having other groups inform of us this first.

Reply:
Also, this is what was causing the specs to fail and cause this PR to be red.  Hence the change in the first place.

Reply:
It's no worse than how I feel about the rest of the `MiqQueue#deliver` calls in the other specs in this PR so I'd rather just get it in before I head out for the day :wink:

Reply:
Was also partially bringing this up since there was a lengthy discussion about this (and similar issues) by the two of us Friday, and I wanted this to be mentioned somewhere public instead of it just being something simply discussed out of band.